### PR TITLE
Splitting GKE examples into two parts, simple & custom

### PR DESCRIPTION
**Pull request by gautambaghel**

### CODE REVIEW

**Comment by gautambaghel** [comment]
@danbarr -- ready for your review

File: examples/tfc-agent-gke-simple/README.md [hunk 1]
```diff
@@ -15,7 +15,6 @@ If running from your own system, you will need:
 - [Terraform](https://developer.hashicorp.com/terraform/downloads)
 - [Google Cloud CLI (`gcloud`)](https://cloud.google.com/sdk/docs/install-sdk)
   - [`gke-gcloud-auth-plugin`](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke)
```
Comment (danbarr) [inline examples/tfc-agent-gke-simple/README.md]: Not needed for the simple example anymore.

```suggestion
```

File: examples/tfc-agent-gke-custom/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    export PROJECT_ID="your-project-id"
+    export LOCATION="us-west1"
+    export REPOSITORY="hashicorp"
+    export IMAGE="tfc-agent"
+    export VERSION="latest"
+
+    # GCP commands to enable services
+    gcloud config set project $PROJECT_ID
+    gcloud services enable cloudbuild.googleapis.com
+    gcloud services enable artifactregistry.googleapis.com
```
Comment (danbarr) [inline examples/tfc-agent-gke-custom/README.md:43]: On my machine, I had to run a `gcloud components update` before the `gcloud artifacts` command would work - might be worth adding here?

File: modules/tfc-agent-gke/variables.tf [hunk 3]
```diff
@@ -132,5 +132,23 @@ variable "tfc_agent_name_prefix" {
...
+variable "tfc_agent_image" {
+  type        = string
+  description = "The Terraform Cloud Agent image to use"
+  default     = "hashicorp/tfc-agent:latest"
+}
+
+variable "tfc_agent_memory_request" {
+  type        = string
+  description = "Memory request for the Terraform Cloud Agent container"
+  default     = "2"
```
Comment (danbarr) [inline modules/tfc-agent-gke/variables.tf]: Two bytes might be a bit light 😉 

```suggestion
  default     = "2Gi"
```

**Review by danbarr — CHANGES_REQUESTED** [review]
Made a few suggestions and one fix, but everything looks great!

File: modules/tfc-agent-gke/variables.tf (hunk 3 above)
Comment (gautambaghel) [reply modules/tfc-agent-gke/variables.tf]: Defaults to 2Gi but explicit is better

Comment (gautambaghel) [reply modules/tfc-agent-gke/variables.tf]: oh maybe not for memory

File: examples/tfc-agent-gke-custom/README.md (hunk 2 above)
Comment (gautambaghel) [reply examples/tfc-agent-gke-custom/README.md:43]: Done :) thanks for the check

File: examples/tfc-agent-gke-simple/README.md (hunk 1 above)
Comment (gautambaghel) [reply examples/tfc-agent-gke-simple/README.md]: Done :) thanks for the check

**Review by danbarr — APPROVED** [review]
LGTM!

**Pull request merged**